src/gluonts/dataset/rolling_dataset.py

```diff
@@ -0,0 +1,197 @@
+) -> Dataset:
+    """
+    Returns an augmented version of the input dataset where each timeseries has 
+    been rolled upon based on the parameters supplied. Below follows an 
+    explanation and examples of how the different parameters can be used to generate
+    differently rolled datasets.
+    
+    The 'rolling' will happen on the data available in the provided window between the 
+    start_time and the end_time for each timeseries. If end_time is omitted, rolling
+    happens on all datapoints from start_time until the end of the timeseries. 
+    The way the data is rolled is governed by the strategy used. 
+
+    Below examples will be based on this one timeseries long dataset
+    [{
+    target=[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]
+    start='2000-1-1-01'
+    }]
+    
+    applying generate_rolling_datasets on this dataset with:
+
+    start_time = pd.Timestamp('2000-1-1-06', '1H')
+    end_time = pd.Timestamp('2000-1-1-10', '1H')
+    strategy=basic_strategy(prediction_length=2)
+
+    returns a new dataset as follows (only target values shown for brevity):
+
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    [1, 2, 3, 4, 5, 6, 7, 8, 9],
+    [1, 2, 3, 4, 5, 6, 7, 8],
+    [1, 2, 3, 4, 5, 6, 7]
+
+    i.e. maximum amount of rolls possible between the end_time and start_time.
+    The basic_strategy only cuts the last value of the target for as long as
+    there is enough values after start_time to perform predictions on.
+
+    When no end time is provided the output is as below since all datapoints
+    from start_time will be rolled over.
+
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14]
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+    [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    [1, 2, 3, 4, 5, 6, 7, 8]
+    [1, 2, 3, 4, 5, 6, 7]
+
+    If the unique_strategy is used, fewer values will be in the output as each
+    roll will be of size prediction_length. This ensures that each prediction
+    will be done on unique/new data. 
+    
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+    [1, 2, 3, 4, 5, 6, 7, 8]
+
+    Not setting an end time and using the unique_strategy results in
+    the below dataset.
+
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
+    [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    [1, 2, 3, 4, 5, 6, 7]
+
+    Parameters
+    ----------
+    dataset
+        Dataset to generate the rolling forecasting datasets from
+    strategy
+        The strategy that is to be used when rolling
+    start_time
+        The start of the window where rolling forecasts should be applied
+    end_time
+        The end time of the window where rolling should be applied
+    Returns
+    -------
+    Dataset 
+        The augmented dataset
```

Comment:
This should be at the top of the docstring.

Reply:
Oki, thought it needed to be in this order, will change it